src/dids/OracleResolveDriver.ts

```diff
@@ -0,0 +1,134 @@
+            }
+            return desiredVal;
+        }
+    }
+
+    /**
+     * Method to query user's config file, which contains ledger specific information
+     * @param option number of which piece of information you need from config file
+     * @summary 
+     * Option 1: CHAINCODE string
+     * Option 2: NETWORK string
+     * Option 3: CHANNEL string
+     * Option 4: ENCODEDCRED string
+     * @returns string of ledger endpoints to hit with REST API calls
+     */
+    public configQuery(option:number, configFileStr:string): string{
+        //check that option is in range 1 to 4
+        if(option < 1 || option > 4 || configFileStr.length == 0){
+            throw new Error("Invalid params passed to configQuery function");
+        }
+
+        //test with mock output
+        if(configFileStr == "TEST"){//give mock output
+            const mockData: string = fs.readFileSync(process.cwd() + "/transcripts/testConfig.txt",'utf-8');
+            return OracleResolveDriver.parseConfig(option, mockData);
+        }
+
+        //actual personal config file query
+        const fileData: string = fs.readFileSync(process.cwd() + configFileStr, 'utf-8');
+        return OracleResolveDriver.parseConfig(option, fileData);
+    }
+
+    /**
+     * Method to use queries config file information and make REST call to associated ledger to get DIDDoc for associated DID
+     * @param did DID to be resolved as string type
+     * @returns Promise of DIDDoc resolution or error type defined in Open-Wallet Credo library
+     */
+    async didResolve(did: string, queryDirectory: string): Promise<any>{
+        if(did.length <= 0 || queryDirectory.length <= 0){
+                if(did.length <= 0){
+                    return -1;
+                }else{
+                    return -1;
+                }
+        }
+        //get did data with passed did and config ledger chaincode like OracleLedgerService line 131
+        var chaincodeStr = this.configQuery(1,queryDirectory);
+        var didDoc = JSON.stringify({
+            chaincode: chaincodeStr,
+            args: ["GetDidDocumentById", did],
+            sync: true,
+        });
+        //hit axios post with network info like OracleLedgerSerivce line 137 and line 148
+        var networkStr = this.configQuery(2,queryDirectory);
+        var channelStr = this.configQuery(3, queryDirectory);
+        var credStr = this.configQuery(4, queryDirectory);
+        let configPost = {
+            url: `${networkStr}/api/v2/channels/${channelStr}/transactions`,
+            method: "post",
+            headers: {
+              Authorization: `Basic ${credStr}`,
+              "Content-Type": "application/json",
+            },
+            data: didDoc,
+        }
+        //wrap axios post in promise data type and return it
+        try {
+            let res = await axios(configPost);
+            return res.data.result.payload;
+          } catch (err) {
+            console.log(err);
```

Comment:
```suggestion
            console.error(err);
```

Reply:
Good note, will make change in next commit